src/app/beer_garden/requests.py

```diff
@@ -677,6 +678,36 @@ def start_request(request_id: str = None, request: Request = None) -> Request:
     return request
 
 
+@publish_event(Events.REQUEST_UPDATED)
+def update_request(
+    request_id: str = None,
+    request: Request = None,
+    new_values=None,
+) -> Request:
+    if new_values is None:
+        new_values = {}
+    request = request or db.query_unique(Request, raise_missing=True, id=request_id)
+    if "expiration_date" in new_values:
+        expiration_date = new_values.pop("expiration_date", None)
+        parent = find_parent(request)
+        parent.expiration_date = expiration_date
+        parent = db.update(parent)
+        if parent.id == request_id:
+            request = parent
+    for key in new_values.keys():
+        setattr(request, key, new_values[key])
+    if new_values:
+        request = db.update(request)
```

Comment:
This seems pretty dangerous. Do we have a use-case in mind driving this change?

Historically we've only allowed updates to requests in certain specific cases, but this seems to expand that. The mongo model validation should prevent some things (changing the output after the request completes, for example). But this still seems dangerous to me - is it now possible to change the ID of a particular request?

Reply:
I added a buffer function for updating `expiration_date` so the function can not be called by the rest api. I initially had it doing just `expiration_date` but changed the function able to work for any attribute to resolve TheBurchLog https://github.com/beer-garden/beer-garden/pull/880#discussion_r583767205 above.